include/range/v3/view/set_algorithm.hpp

```diff
@@ -0,0 +1,904 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Eric Niebler 2013-2014
+//  Copyright Tomislav Ivek 2015-2016
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_VIEW_SET_ALGORITHM_HPP
+#define RANGES_V3_VIEW_SET_ALGORITHM_HPP
+
+#include <utility>
+#include <iterator>
+#include <algorithm>
+#include <functional>
+#include <type_traits>
+#include <meta/meta.hpp>
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/size.hpp>
+#include <range/v3/begin_end.hpp>
+#include <range/v3/range_traits.hpp>
+#include <range/v3/view_facade.hpp>
+#include <range/v3/utility/move.hpp>
+#include <range/v3/utility/semiregular.hpp>
+#include <range/v3/utility/functional.hpp>
+#include <range/v3/utility/static_const.hpp>
+#include <range/v3/view/view.hpp>
+#include <range/v3/view/all.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        /// \cond
+        namespace detail
+        {
+            // indirect_move is put here instead of in set_??_view::cursor to
+            // work around gcc friend name injection bug.
+            template<typename Cursor>
+            struct set_op_cursor_move
+            {
+                template<typename Sent>
+                friend auto indirect_move(basic_iterator<Cursor, Sent> const &it)
+                RANGES_DECLTYPE_AUTO_RETURN_NOEXCEPT
+                (
+                    get_cursor(it).indirect_move_()
+                )
+            };
+            
+            template<typename Rng1, typename Rng2,
+                     typename C, typename P1, typename P2,
+                     template<bool, typename...> class Cursor, cardinality Cardinality>
+            struct set_algorithm_view
+            : view_facade<
+                    set_algorithm_view<Rng1, Rng2, C, P1, P2, Cursor, Cardinality>,
+                    Cardinality>
+            {
+            private:
+                friend range_access;
+                semiregular_t<function_type<C>> pred_;
+                semiregular_t<function_type<P1>> proj1_;
+                semiregular_t<function_type<P2>> proj2_;
+                Rng1 rng1_;
+                Rng2 rng2_;
+
+                struct sentinel;
```

Comment:
... And it is not needed, the default sentinel is supposed to be used. Will remove it.